src/beamlime/logging/formatters.py

```diff
@@ -0,0 +1,156 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+from collections import OrderedDict
+from functools import partial
+from logging import Formatter
+from typing import Any, Literal, Mapping, TypeAlias, overload
+
+_FormatStyle: TypeAlias = Literal["{"]
+_SepStyle: TypeAlias = Literal["|", ","]
+
+
+class LogColumn:
+    def __init__(
+        self,
+        variable_name: str,
```

Comment:
In Scipp, we require keyword arguments for functions with easily confused arguments. E.g. multiple str-args. You should probably do this, too.